web-ui/src/main/resources/catalog/components/utility/CORSInterceptor.js

```diff
@@ -133,6 +133,14 @@
                   gnGlobalSettings.gnUrl + (gnLangs.current || "eng") + "/" + config.url;
               }
 
+              /*
+              Reset requireProxy array. This prevent same host of map server and catalouge deployment affecting together.
+              For example: One online resource of wms https://example.com/mapserver/ows?service=WMS&request=GetCapabilities&layers=roads
+              and another online resource of file store https://example.com/geonetwork/srv/api/records/c90c284d-171f-41a9-bc92-b2533970cdc8/attachments/sample.txt
+              The will confuse the proxy to think the filestore has same hostname and will use proxy url. But due to the staight security meansure the proxy url will not work for the backend store
+              */
+              gnGlobalSettings.requireProxy = [];
```

Comment:
I'm not really sure about this change. Also not very familiar with this code. 

If I'm not misunderstanding the change it seems resetting **all** the values in `gnGlobalSettings.requireProxy` when the configuration for the URL requested is created.

So for example if `gnGlobalSettings.requireProxy` contains 

```
[
   "https://example.com/#GET",
   "https://anotherdomain.com/#GET"
]
```

You are also removing the "https://anotherdomain.com/#GET" when accessing to the filestore in https://example.com/...

I'm not sure this is correct, as in that case, which is the purpose to `gnGlobalSettings.requireProxy`

Reply:
yes, both domain will get removed. I am not sure what the benefit to cache the domain. The list can grow unlimitedly. Also in our case, one wms host cached, it will bring problem to the file store host which should not go thru the proxy